be/src/storage/sstable/format.cpp

```diff
@@ -125,20 +166,38 @@ Status ReadBlock(RandomAccessFile* file, const ReadOptions& options, const Block
         size_t ulength = 0;
         if (!snappy::GetUncompressedLength(data, n, &ulength)) {
             delete[] buf;
-            return Status::Corruption("corrupted compressed block contents");
+            return Status::Corruption("corrupted compressed block (snappy) contents");
         }
         char* ubuf = new char[ulength];
         if (!snappy::RawUncompress(data, n, ubuf)) {
             delete[] buf;
             delete[] ubuf;
-            return Status::Corruption("corrupted compressed block contents");
+            return Status::Corruption("corrupted compressed block (snappy) contents");
         }
         delete[] buf;
         result->data = Slice(ubuf, ulength);
         result->heap_allocated = true;
         result->cachable = true;
         break;
     }
+    case kLz4FrameCompression: {
+        auto st = DecompressBlock(CompressionTypePB::LZ4_FRAME, data, n, result);
+        delete[] buf;
+        if (!st.ok()) {
+            LOG(ERROR) << st.to_string();
+            return Status::Corruption("corrupted compressed block (lz4) contents");
+        }
+        break;
+    }
+    case kZstdCompression: {
+        auto st = DecompressBlock(CompressionTypePB::ZSTD, data, n, result);
+        delete[] buf;
+        if (!st.ok()) {
+            LOG(ERROR) << st.to_string();
+            return Status::Corruption("corrupted compressed block (zstd) contents");
+        }
+        break;
+    }
```

Comment:
better add the compress type in line 203. so in case of rollback in future, still possible known the compression type even it is not supported.

Reply:
done